### PR TITLE
Cmake fixes for opencv, tbb and png

### PR DESCRIPTION
**Pull request by fabiencastan**

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -132,24 +128,29 @@ else(WITH_CUDA)
...
+                           PRIVATE ${TBB_DEFINITIONS}) 
   
-  target_link_libraries(CCTag 
+target_link_libraries(CCTag 
                       PUBLIC ${OpenCV_LIBS} ${Boost_LIBRARIES} ${Eigen_LIBRARIES} 
-                      PRIVATE ${TBB_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES} pthread dl)
+                      PRIVATE ${TBB_LIBRARIES} ${JPEG_LIBRARIES} pthread dl)
 
-endif(WITH_CUDA)
+if(HAVE_PNG)
```
Comment (simogasp) [inline src/CMakeLists.txt]: maybe `PNG_FOUND` ? I can't see any definition for `HAVE_PNG`
Also, I suspect that PNG is only needed by opencv, if it is compiled with. It seems to me that the png library it is not explicitly used anywhere in the code
And we can safely remove these two lines
https://github.com/alicevision/CCTag/blob/develop/src/applications/simulation/main.cpp#L8

@griwodz

Comment (fabiencastan) [inline src/CMakeLists.txt]: Yes, good point.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -132,24 +128,24 @@ else(WITH_CUDA)
     set_target_properties(CCTag PROPERTIES VERSION ${PROJECT_VERSION})
   endif( COMPILE_STATIC_CCTAG_LIBRARY )
 
-  target_include_directories(CCTag 
+endif(WITH_CUDA)
```
Comment (simogasp) [inline src/CMakeLists.txt]: nope, you cannot do that, "share the code", as CUDA does not yet support the PUBLIC/PRIVATE stuff yet, it uses the old-fashion include_directories etc.

Comment (fabiencastan) [inline src/CMakeLists.txt]: Ok, I was not aware of that.
Updated.

**Pull request merged**